### DISCUSSION TITLE
[Run issue]:Problem using Cmake at step 8.4 on WSL

### DISCUSSION THREAD

**Discussion by Strong-Yang-coder** [OpenTOPAS/OpenTOPAS#73] [Run-Problem] [answered]
### What was the issue?

I follow the QuckStart Guide for WSL . But in step 8.4 there are two errors.
strong@StrongGo:~/Applications/TOPAS/OpenTOPAS-build$ make -j10 install
[  1%] Automatic MOC for target variance
[  1%] Built target variance_autogen
[  1%] Building CXX object variance/CMakeFiles/variance.dir/variance_autogen/mocs_compilation.cpp.o
[  2%] Building CXX object variance/CMakeFiles/variance.dir/TsAutomaticImportanceSamplingManager.cc.o
[  2%] Building CXX object variance/CMakeFiles/variance.dir/TsAutomaticImportanceSamplingOperator.cc.o
[  2%] Building CXX object variance/CMakeFiles/variance.dir/TsAutomaticImportanceSamplingParallelManager.cc.o
[  3%] Building CXX object variance/CMakeFiles/variance.dir/TsAutomaticImportanceSamplingParallelOperator.cc.o
[  3%] Building CXX object variance/CMakeFiles/variance.dir/TsAutomaticImportanceSamplingParallelProcess.cc.o
[  3%] Building CXX object variance/CMakeFiles/variance.dir/TsAutomaticImportanceSamplingProcess.cc.o
[  3%] Building CXX object variance/CMakeFiles/variance.dir/TsCrossSectionEnhancement.cc.o
[  4%] Building CXX object variance/CMakeFiles/variance.dir/TsDirectionalRussianRoulette.cc.o
In file included from /home/strong/Applications/TOPAS/OpenTOPAS/variance/TsDirectionalRussianRoulette.cc:33:
/home/strong/Applications/TOPAS/OpenTOPAS/geometry/TsVGeometryComponent.hh:52:7: error: using typedef-name ‘using G4VTouchable = class G4TouchableHistory’ after ‘class’
   52 | class G4VTouchable;
      |       ^~~~~~~~~~~~
In file included from /home/strong/Applications/GEANT4/geant4-install/include/Geant4/G4TouchableHandle.hh:42,
                 from /home/strong/Applications/GEANT4/geant4-install/include/Geant4/G4Track.hh:55,
                 from /home/strong/Applications/TOPAS/OpenTOPAS/variance/TsDirectionalRussianRoulette.hh:37,
                 from /home/strong/Applications/TOPAS/OpenTOPAS/variance/TsDirectionalRussianRoulette.cc:31:
/home/strong/Applications/GEANT4/geant4-install/include/Geant4/G4VTouchable.hh:39:7: note: ‘using G4VTouchable = class G4TouchableHistory’ has a previous declaration here
   39 | using G4VTouchable = G4TouchableHistory;
      |       ^~~~~~~~~~~~
make[2]: *** [variance/CMakeFiles/variance.dir/build.make:188: variance/CMakeFiles/variance.dir/TsDirectionalRussianRoulette.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1177: variance/CMakeFiles/variance.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
strong@StrongGo:~/Applications/TOPAS/OpenTOPAS-build$ cmake ../OpenTOPAS -DCMAKE_INSTALL_PREFIX=../OpenTOPAS-install
-- CMAKE_CXX_COMPILER_ID: GNU
-- CMAKE_CXX_COMPILER_VERSION: 11.4.0
CMake Warning at cmake/Modules/TOPAS_Options.cmake:9 (message):
  Setting CMAKE_BUILD_TYPE=Release
Call Stack (most recent call first):
  CMakeLists.txt:21 (include)

-- Found Geant4 v11.2 (patch 2)
-- Found GDCM v2.6
-- Prepending the following flags to CXX_FLAGS =
-- (BASICS)   -Wl,--disable-new-dtags
-- (WARNINGS) -Wall -Wextra -Wpedantic -Wmaybe-uninitialized
-- (RELEASE)  -O3 -DNDEBUG -fno-trapping-math -ftree-vectorize -fno-math-errno
-- CMAKE_BUILD_TYPE: Release
-- TOPAS_MT: ON
-- TOPAS_WITH_STATIC_GEANT4: OFF
-- TOPAS_COVERAGE: OFF
-- TOPAS_INSTALL_EXAMPLES: ON
-- TOPAS-VERSION: 4.0.0
-- Configuring done
-- Generating done
-- Build files have been written to: /home/strong/Applications/TOPAS/OpenTOPAS-build
strong@StrongGo:~/Applications/TOPAS/OpenTOPAS-build$ sudo make -j1 install
[  1%] Automatic MOC for target variance
[  1%] Built target variance_autogen
Consolidate compiler generated dependencies of target variance
[  2%] Building CXX object variance/CMakeFiles/variance.dir/TsDirectionalRussianRoulette.cc.o
In file included from /home/strong/Applications/TOPAS/OpenTOPAS/variance/TsDirectionalRussianRoulette.cc:33:
/home/strong/Applications/TOPAS/OpenTOPAS/geometry/TsVGeometryComponent.hh:52:7: error: using typedef-name ‘using G4VTouchable = class G4TouchableHistory’ after ‘class’
   52 | class G4VTouchable;
      |       ^~~~~~~~~~~~
In file included from /home/strong/Applications/GEANT4/geant4-install/include/Geant4/G4TouchableHandle.hh:42,
                 from /home/strong/Applications/GEANT4/geant4-install/include/Geant4/G4Track.hh:55,
                 from /home/strong/Applications/TOPAS/OpenTOPAS/variance/TsDirectionalRussianRoulette.hh:37,
                 from /home/strong/Applications/TOPAS/OpenTOPAS/variance/TsDirectionalRussianRoulette.cc:31:
/home/strong/Applications/GEANT4/geant4-install/include/Geant4/G4VTouchable.hh:39:7: note: ‘using G4VTouchable = class G4TouchableHistory’ has a previous declaration here
   39 | using G4VTouchable = G4TouchableHistory;
      |       ^~~~~~~~~~~~
make[2]: *** [variance/CMakeFiles/variance.dir/build.make:188: variance/CMakeFiles/variance.dir/TsDirectionalRussianRoulette.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1177: variance/CMakeFiles/variance.dir/all] Error 2
make: *** [Makefile:166: all] Error 2

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by JoseRamosMendez** [comment, marked as answer]
Hi there! 
It looks like you're using a different version of Geant4 (). Would you please try Geant4 version 11.1.3 (or 11.1.2).
Best

**Reply by rrdipto137** [reply]
I have same problem using the latest version.but still
